src/runtime/virtcontainers/kata_agent.go

```diff
@@ -394,13 +394,27 @@ func setupSandboxBindMounts(sandbox *Sandbox) error {
 	if err := os.MkdirAll(sandboxMountDir, DirMode); err != nil {
 		return fmt.Errorf("Creating sandbox shared mount directory: %v: %w", sandboxMountDir, err)
 	}
+	var mountedList []string
+	defer func() {
+		if err != nil {
+			for _, mnt := range mountedList {
+				if err := syscall.Unmount(mnt, syscall.MNT_DETACH|UmountNoFollow); err != nil {
+					k.Logger().WithError(err).Errorf("cleanup: couldn't unmount %s", mnt)
+				}
+			}
+			if err := os.RemoveAll(sandboxMountDir); err != nil {
```

Comment:
And the same here.

I noticed you use `retErr` in other functions, I'd go for the same on this one.